src/peft/optimizers/loraplus.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains the implementation of the LoraPlus optimizer.
+"""
+
+from __future__ import annotations
+
+from operator import attrgetter
+
+import torch.nn as nn
+from torch.optim import Optimizer
+from transformers.pytorch_utils import ALL_LAYERNORM_LAYERS
+from transformers.trainer_pt_utils import get_parameter_names
+
+from ..peft_model import PeftModel
+from ..tuners.lora.layer import Embedding
+
+
+def create_loraplus_optimizer(
+    model: PeftModel,
+    optimizer_cls: type[Optimizer],
+    optimizer_kwargs: dict,
```

Comment:
A suggestion: Let's remove `optimizer_kwargs` and just add `**kwargs`. IMO, that makes calling this function easier, as we can use `create_loraplus_optimizer(..., weight_decay=1e-3)` instead of `create_loraplus_optimizer(..., optimizer_kwargs={..., "weight_decay": 1e-3})`. And since `lr` is not optional, let's make this a normal arg of `create_loraplus_optimizer`.

Reply:
Good point.